llvm/lib/Target/AArch64/AArch64InstrInfo.cpp

```diff
@@ -3918,21 +3924,21 @@ bool AArch64InstrInfo::getMemOpInfo(unsigned Opcode, TypeSize &Scale,
   case AArch64::LD1RSH_S_IMM:
   case AArch64::LD1RSH_D_IMM:
     Scale = TypeSize::getFixed(2);
-    Width = 2;
+    Width = TypeSize::getFixed(2);
     MinOffset = 0;
     MaxOffset = 63;
     break;
   case AArch64::LD1RW_IMM:
   case AArch64::LD1RW_D_IMM:
   case AArch64::LD1RSW_IMM:
     Scale = TypeSize::getFixed(4);
```

Comment:
aren't LD1RSW* SVE instructions as well? Any reasons why we don't change them to getScalable?

Reply:
I believe they load a scalar and broadcast them to all lanes, so only load a fixed number of bytes is read from memory.

Reply:
ah ok yeah makes  sense